mesonbuild/interpreter/interpreter.py

```diff
@@ -603,35 +623,25 @@ def _import_module(self, modname: str, required: bool, node: mparser.BaseNode) -
             FeatureNew.single_use(f'module {ext_module.INFO.name}', ext_module.INFO.added, self.subproject, location=node)
         if ext_module.INFO.deprecated:
             FeatureDeprecated.single_use(f'module {ext_module.INFO.name}', ext_module.INFO.deprecated, self.subproject, location=node)
+        if expect_unstable and not ext_module.INFO.unstable and ext_module.INFO.stabilized is None:
+            raise InvalidArguments(f'Module {modname} has never been unstable, remove "unstable-" prefix.')
+        if ext_module.INFO.stabilized:
+            FeatureDeprecated.single_use(
+                f'module {ext_module.INFO.name} has been stabilized',
+                ext_module.INFO.stabilized, self.subproject,
+                'drop "unstable-" prefix from the module name',
+                location=node)
+            FeatureNew.single_use(
+                f'module {ext_module.INFO.name} as stable module',
+                ext_module.INFO.stabilized, self.subproject,
+                f'Consider either adding "unstable-" to the module name, or updating the meson required version to ">= {ext_module.INFO.stabilized}"',
+                location=node)
```

Comment:
This needs to be if/else expect_unstable?

Reply:
These both need to be triggered in the stablized case, the first applies when the targetted meson is 0.60, the module was stabilized in 0.62, but it's being imported as `import('module')` instead of `import('unstable-module')`, since that would fail with 0.60.

The second is needed when your require 0.62, but are still importing as `import('unstable-module')`.

Do either of those need to be  triggered in additonal cases?

Reply:
expect_unstable is not the same as stabilized though.

The current version afaict will print a warning, if:
- you use `import('unstable-module')`, that you should "add unstable-", and
- you use `import('module'), that it's been stabilized and you should drop the "unstable-"

This would be fixed by only printing the featurenew if stabilized ***and*** not expect_unstable, and only printing the featuredeprecated if stabilized ***and*** expect_unstable.

Reply:
Right, I've fixed this in the most recent version.